components/ModEditSubmissionForm.vue

```diff
@@ -513,36 +513,40 @@ const router = useRouter()
 const modalStore = useModalStore()
 const screenStore = useModerationScreenStore()
 
+const submissionFormFields = {
 // contactFields
-const nameEn: Ref<string> = ref('')
-const nameJa: Ref<string> = ref('')
-const phone: Ref<string> = ref('')
-const website: Ref<string> = ref('')
-const email: Ref<string> = ref('')
-// addressesFields
-const postalCode: Ref<string> = ref('')
-const prefectureEn: Ref<string> = ref('')
-const cityEn: Ref<string> = ref('')
-const addressLine1En: Ref<string> = ref('')
-const addressLine2En: Ref<string> = ref('')
-const prefectureJa: Ref<string> = ref('')
-const cityJa: Ref<string> = ref('')
-const addressLine1Ja: Ref<string> = ref('')
-const addressLine2Ja: Ref<string> = ref('')
-// googleMapsFields
-const googlemapsURL: Ref<string> = ref('')
-const mapLatitude: Ref<string> = ref('')
-const mapLongitude: Ref<string> = ref('')
-//healthcareProfessionalFields
-const healthCareProfessionalNameArray: Ref<Array<LocalizedNameInput>> = ref([])
-const localizedFirstName: Ref<string> = ref('')
-const localizedLastName: Ref<string> = ref('')
-const localizedMiddleName: Ref<string> = ref('')
-const nameLocale: Ref<Locale> = ref(Locale.EnUs)
-const healthcareProfessionalAcceptedInsurances: Ref<Array<Insurance>> = ref([])
-const healthcareProfessionalDegrees: Ref<Array<Degree>> = ref([])
-const healthcareProfessionalSpecialties: Ref<Array<Specialty>> = ref([])
-const healthcareProfessionalLocales: Ref<Array<Locale>> = ref([])
+    nameEn: ref('') as Ref<string>,
+    nameJa: ref('') as Ref<string>,
+    phone: ref('') as Ref<string>,
+    website: ref('') as Ref<string>,
+    email: ref('') as Ref<string>,
+    // addressesFields
+    postalCode: ref('') as Ref<string>,
+    prefectureEn: ref('') as Ref<string>,
+    cityEn: ref('') as Ref<string>,
+    addressLine1En: ref('') as Ref<string>,
+    addressLine2En: ref('') as Ref<string>,
+    prefectureJa: ref('') as Ref<string>,
+    cityJa: ref('') as Ref<string>,
+    addressLine1Ja: ref('') as Ref<string>,
+    addressLine2Ja: ref('') as Ref<string>,
+    // googleMapsFields
+    googlemapsURL: ref('') as Ref<string>,
+    mapLatitude: ref('') as Ref<string>,
+    mapLongitude: ref('') as Ref<string>,
+    //healthcareProfessionalFields
+    healthCareProfessionalNameArray: ref([]) as Ref<Array<LocalizedNameInput>>,
+    localizedFirstName: ref('') as Ref<string>,
+    localizedLastName: ref('') as Ref<string>,
+    localizedMiddleName: ref('') as Ref<string>,
+    nameLocale: ref(Locale.EnUs) as Ref<Locale>,
+    healthcareProfessionalAcceptedInsurances: ref([]) as Ref<Array<Insurance>>,
+    healthcareProfessionalDegrees: ref([]) as Ref<Array<Degree>>,
+    healthcareProfessionalSpecialties: ref([]) as Ref<Array<Specialty>>,
+    healthcareProfessionalLocales: ref([]) as Ref<Array<Locale>>
+} as { [key: string]: Ref }
+
+const submissionBeforeChanges: { [key: string]: (string | []) } = {}
```

Comment:
These fields were moved into an object to make it easier to determine if changes were made to the form.

Reply:
💯 awesome!